core/src/banking_stage.rs

```diff
@@ -1460,6 +1466,36 @@ impl BankingStage {
                         .batched_execute_cost
                         .saturating_add(cost.execution_cost);
                 }
+                Err(transaction_error) => match transaction_error {
+                    TransactionError::WouldExceedMaxBlockCostLimit => {
+                        cost_details.batched_retried_txs_per_block_limit_count = cost_details
+                            .batched_retried_txs_per_block_limit_count
+                            .saturating_add(1)
+                    }
+                    TransactionError::WouldExceedMaxVoteCostLimit => {
+                        cost_details.batched_retried_txs_per_vote_limit_count = cost_details
+                            .batched_retried_txs_per_vote_limit_count
+                            .saturating_add(1)
+                    }
+                    TransactionError::WouldExceedMaxAccountCostLimit => {
+                        cost_details.batched_retried_txs_per_account_limit_count = cost_details
+                            .batched_retried_txs_per_account_limit_count
```

Comment:
can use the `saturating_add_assign!` macro for these calls

Reply:
Done!